en/companion_computer/holybro_pixhawk_jetson_baseboard.md

```diff
@@ -0,0 +1,1143 @@
+Assuming you're using a Pixhawk 6x, the command would be:
+
+```sh
+make px4_fmu-v6x_default
+```
+
+If this builds successfully, you can connect the Pixhawk USB-C port to the Jetson USB port as shown below (the USB cable comes with the kit).
+
+![How to connect Jetson to Pixhawk directly](../../assets/companion_computer/holybro_pixhawk_jetson_baseboard/upload_px4_connection.png)
+
+Then upload the firmware to Pixhawk by building with the `upload` argument:
+
+```sh
+make px4_fmu-v6x_default upload
+```
+
+## ROS 2 Installation and Setup
+
+Follow the instruction in the ROS2 User guide to [Install ROS 2](../ros2/user_guide.md#install-ros-2) on the Jetson.
+You don't have to install PX4 on the Jetson because we're not using the simulator, but you may wish to install the full desktop so you can use additional ROS 2 packages for further development:
+
+```sh
+sudo apt install -y ros-humble-desktop-full -y
+```
+
+Then fetch and build the uXRCE-DDS _agent_ as covered in [Setup the Agent](../ros2/user_guide.md#setup-the-agent) in the ROS 2 User guide.
+We don't need to start the agent yet, because the client is not yet running.
+
+## Ethernet Setup using Netplan
+
+Next we set up an Ethernet connection between the Jetson and PX4 that will be used by our ROS 2 client and agent to communicate.
+
+::: warning
+The Holybro Jetson Pixhawk carrier has an internal network switch between Pixhawk and Jetson, so you do not need to connect any external cables.
+:::
+
+These instructions approximately mirror the [PX4 Ethernet setup](../advanced_config/ethernet_setup.md) (and [Holybro Pixhawk RPi CM4 Baseboard](../companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md)).
+
+1. Modify the Netplan configuration file to set up a static IP for the Jetson.
+   You can usually find the Netplan configuration file in the `/etc/netplan/` directory.
+   It's typically named something like `01-netcfg.yaml`, but the name can vary.
+
+   Open the file in your preferred text editor:
+
+   ```sh
+   sudo nano /etc/netplan/01-netcfg.yaml
+   ```
+
+2. Modify the file to configure the Ethernet interface:
+
+   ```sh
+   network:
+   version: 2
+   renderer: networkd
+   ethernets:
+     eth0:
+         dhcp4: no
+         addresses:
+         - 192.168.0.2/24
+   ```
+
+   This gives the Jetson a static IP address on the Ethernet of `192.168.0.2`
+
+3. Apply the Netplan configuration:
+
+   Save the file and apply the changes using the following command:
+
+   ```sh
+   sudo netplan apply
+   ```
+
+The Pixhawk Ethernet address is set to `192.168.0.3`.
+
+We ping Pixhawk now inside Jetson terminal:
+
+```sh
+ping 192.168.0.3
+```
+
+If this is successful you will see the output below.
+
+![Pixhawk and Jetson successful ping](../../assets/companion_computer/holybro_pixhawk_jetson_baseboard/pixhawk_ping.png)
+
+This means you can run your XRCE-DDS agent on Jetson and have your ROS nodes talk to the client running on PX4.
+
+## Connection Sanity Check between Pixhawk and Jetson
```

Comment:
@farhangnaderi Would I be correct that you are just doing this to show that the serial port works? You never disable mavlink later on, so you can't use it for XRCE-DDS. I'll pull it out of the flow later.